airflow/migrations/versions/88537a2fcad_change_uniqueconstraint_for_dagrun.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""change uniqueconstraint for dagrun
+
+Revision ID: 88537a2fcad
+Revises: 182c0d1598ee
+Create Date: 2016-04-27 23:21:26.574027
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '88537a2fcad'
+down_revision = '182c0d1598ee'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+from alembic import context
+import sqlalchemy as sa
+
+
+def upgrade():
+    url = context.config.get_main_option("sqlalchemy.url")
+    if url.find("postgresql") > -1:
+        op.drop_constraint("dag_run_dag_id_execution_date_key", "dag_run")
+        op.create_unique_constraint("uq_dag_run_dag_id_execution_date_run_id",
+                                    "dag_run", ["dag_id", "execution_date", "run_id"])
+    elif url.find("mysql") > -1:
+        op.drop_constraint("dag_id", "dag_run", "unique")
+        op.create_unique_constraint("uq_dag_run_dag_id_execution_date_run_id",
+                                    "dag_run", ["dag_id", "execution_date", "run_id"])
+    elif url.find("sqlite") > -1:
```

Comment:
Also, I am running into an unrelated bug that required me to delete a unique constraint called "dag_id_2" for mysql ONLY. I don't understand why I had to do that, or where it came from, or -- most bizarre -- why it only decided to pop up in 1 unit test even though the code is used in every test... but just in case it helps you, I had to use your code with that constraint name.